src/Spatial/Euclidean/StraightLine.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace MathNet.Spatial.Euclidean
+{
+    /// <summary>
+    /// Represents a line conforming to the equation y = mx + b where m is the gradient and b is the y-intercept
+    /// </summary>
+    public class StraightLine : ILine
+    {
+        private double yoffset;
+        private double xoffset;
+
+        /// <summary>
+        /// The gradient of the line
+        /// </summary>
+        public double Gradient { get; }
+
+        /// <summary>
+        /// Creates a Line conforming to the equation y = mx + b where point is a point on the line and gradient is the slope of the line
+        /// </summary>
+        /// <param name="point">A point on the line</param>
+        /// <param name="gradient">The gradient of the line</param>
+        public StraightLine(Point2D point, double gradient)
+        {
+            Gradient = gradient;
+            setOffset(point.X, point.Y);
+        }
+
+        /// <summary>
+        /// Creates a Line conforming to the equation y = mx + b which passes through points p1 and p2
+        /// </summary>
+        /// <param name="p1">A point on the line</param>
+        /// <param name="p2">A point on the line</param>
+        public StraightLine(Point2D p1, Point2D p2)
+        {
+            if (p1 == p2)
+                throw new ArgumentException("A line cannot be formed from two identical points");
+            Gradient = (p2.Y - p1.Y) / (p2.X - p1.X);
+            setOffset(p1.X, p1.Y);
+        }
+
+        private void setOffset(double x, double y)
```

Comment:
I prefer `private static void SetOffset(ref double x, ref double y)`. That allows the fields to be `readonly`.
Rename to `SetOffset`, also private methods last according to stylecop.